File: example/java_ls/config/other/Loot.java
package config.other;

public class Loot {
    private int lootid;
    private String ename;
    private config.Text name;
    private java.util.List<Integer> chanceList;
    private java.util.List<config.other.Lootitem> ListRefLootid;
    private java.util.List<config.other.Lootitem> ListRefAnotherWay;

    private Loot() {
    }

    public static Loot _create(configgen.genjava.ConfigInput input) {
        Loot self = new Loot();
        self.lootid = input.readInt();
        self.ename = input.readStringInPool();
        self.name = config.Text._create(input);
        {
            int c = input.readInt();
            if (c == 0) {
                self.chanceList = java.util.Collections.emptyList();
            } else {
                self.chanceList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.chanceList.add(input.readInt());
                }
            }
        }
        return self;
    }

    /**
     * 序号
     */
    public int getLootid() {
        return lootid;
    }

    public String getEname() {
        return ename;
    }

    /**
     * 名字
     */
    public config.Text getName() {
        return name;
    }

    /**
     * 掉落0件物品的概率
     */
    public java.util.List<Integer> getChanceList() {
        return chanceList;
    }

    public java.util.List<config.other.Lootitem> listRefLootid() {
        return ListRefLootid;
    }

    public java.util.List<config.other.Lootitem> listRefAnotherWay() {
        return ListRefAnotherWay;
    }

    @Override
    public String toString() {
        return "(" + lootid + "," + ename + "," + name + "," + chanceList + ")";
    }

    public void _resolveDirect(config.ConfigMgr mgr) {
        ListRefLootid = new java.util.ArrayList<>();
        for (config.other.Lootitem v : mgr.other_lootitem_All.values()) {
            if (v.getLootid() == lootid)
                ListRefLootid.add(v);
        }
        ListRefLootid = ListRefLootid.isEmpty() ? java.util.Collections.emptyList() : new java.util.ArrayList<>(ListRefLootid);
        ListRefAnotherWay = new java.util.ArrayList<>();
        for (config.other.Lootitem v : mgr.other_lootitem_All.values()) {
            if (v.getLootid() == lootid)
                ListRefAnotherWay.add(v);
        }
        ListRefAnotherWay = ListRefAnotherWay.isEmpty() ? java.util.Collections.emptyList() : new java.util.ArrayList<>(ListRefAnotherWay);
    }

    public void _resolve(config.ConfigMgr mgr) {
        _resolveDirect(mgr);
    }

    public static Loot get(int lootid) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getOtherLoot(lootid);
    }

    public static java.util.Collection<Loot> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allOtherLoot();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Loot self = Loot._create(input);
                mgr.other_loot_All.put(self.lootid, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Loot e : mgr.other_loot_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}

File: example/lua_ls_client/cfg/other/loot.lua
local cfg = require "cfg._cfgs"

---@class cfg.other.loot
---@field lootid number , 序号
---@field ename string 
---@field name text , 名字
---@field chanceList table<number,number> , 掉落0件物品的概率
---@field get fun(lootid:number):cfg.other.loot
---@field all table<any,cfg.other.loot>
---@field ListRefLootid table<number,cfg.other.lootitem>
---@field ListRefAnotherWay table<number,cfg.other.lootitem>

local this = cfg.other.loot

local mk = cfg._mk.i18n_table(this, { { 'all', 'get', 1 }, }, nil, { 
    { 'ListRefLootid', 2, cfg.other.lootitem, 'all', 1, 1 }, 
    { 'ListRefAnotherWay', 2, cfg.other.lootitem, 'all', 1, 1 }, }, 
    { name = 1 },
    'lootid', -- int, 序号
    'ename', -- str
    'name', -- text, 名字
    'chanceList' -- list<int>, 掉落0件物品的概率
    )

local A = {}
A[1] = {0, 100, 0, 0, 0, 0, 0}
A[2] = {80, 20, 0, 0, 0, 0, 0}

mk(1, "", 14, {100, 200, 200, 200, 200, 50, 50})
mk(2, "combo1", 15, A[1])
mk(3, "combo2", 16, A[1])
mk(4, "combo3", 17, A[1])
mk(5, "", 18, {20, 10, 10, 20, 20, 10, 10})
mk(6, "", 19, A[1])
mk(7, "", 20, A[1])
mk(8, "", 21, A[2])
mk(9, "", 22, A[2])
mk(10, "", 23, A[2])

return this
